Index: share/make/basepkg.mk
===================================================================
--- share/make/basepkg.mk	(revision 67410)
+++ share/make/basepkg.mk	(working copy)
@@ -85,15 +85,25 @@
 
 mkdesc:
 	@if test -f DESCRIPTION; then \
+	  if test "$(PKG_BUILT_STAMP)" != ""; then \
+	    $(ECHO) "tools:::.install_package_description('.', '$(top_builddir)/library/${pkg}', '$(PKG_BUILT_STAMP)')" | \
+	    R_DEFAULT_PACKAGES=NULL $(R_EXE) > /dev/null ; \
+	  else \
 	  $(ECHO) "tools:::.install_package_description('.', '$(top_builddir)/library/${pkg}')" | \
 	  R_DEFAULT_PACKAGES=NULL $(R_EXE) > /dev/null ; \
+	  fi; \
 	fi
 
 ## for base and tools
 mkdesc2:
 	@$(INSTALL_DATA) DESCRIPTION $(top_builddir)/library/$(pkg)
-	@$(ECHO) "Built: R $(VERSION); ; `TZ=UTC date`; $(R_OSTYPE)" \
-	   >> $(top_builddir)/library/$(pkg)/DESCRIPTION
+	@if test "$(PKG_BUILT_STAMP)" != ""; then \
+	  $(ECHO) "Built: R $(VERSION); ; $(PKG_BUILT_STAMP); $(R_OSTYPE)" \
+	     >> $(top_builddir)/library/$(pkg)/DESCRIPTION ; \
+	else \
+	  $(ECHO) "Built: R $(VERSION); ; `TZ=UTC date`; $(R_OSTYPE)" \
+	     >> $(top_builddir)/library/$(pkg)/DESCRIPTION ; \
+	fi
 
 mkdemos:
 	@$(ECHO) "tools:::.install_package_demos('$(srcdir)', '$(top_builddir)/library/$(pkg)')" | \
